tgui/packages/tgui/interfaces/PowerTransmissionLaser.js

```diff
@@ -0,0 +1,215 @@
+          {formatSiUnit(charge, 0, 'J')}
+        </LabeledList.Item>
+      </LabeledList>
+      <ProgressBar
+        mt="0.5em"
+        mb="0.5em"
+        ranges={{
+          good: [0.8, Infinity],
+          average: [0.5, 0.8],
+          bad: [-Infinity, 0.5],
+        }}
+        value={charge / capacity} />
+      <LabeledList>
+        <LabeledList.Item label="Grid Saturation" />
+      </LabeledList>
+      <ProgressBar
+        mt="0.5em"
+        ranges={{
+          good: [0.8, Infinity],
+          average: [0.5, 0.8],
+          bad: [-Infinity, 0.5],
+        }}
+        value={gridLoad / totalGridPower} />
+    </Section>
+  );
+};
+
+const InputControls = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    chargingEnabled,
+    excessPower,
+    isCharging,
+    inputLevel,
+    inputNumber,
+    inputMultiplier,
+  } = data;
+
+  return (
+    <Section title="Input Controls">
+      <LabeledList>
+        <LabeledList.Item
+          label="Input Circuit"
+          buttons={
+            <Button
+              icon="power-off"
+              content={chargingEnabled ? 'Enabled' : 'Disabled'}
+              color={chargingEnabled ? 'green' : 'red'}
+              onClick={() => act('toggleInput')} />
+          } >
+          <Box
+            color={(isCharging && 'good')
+              || (chargingEnabled && 'average')
+              || 'bad'}>
+            {(isCharging && 'Online')
+              || (chargingEnabled && 'Idle')
+              || 'Offline'}
+          </Box>
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Input Level" >
+          {formatPower(inputLevel)}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Optimal" >
+          {formatPower(excessPower)}
+        </LabeledList.Item>
+      </LabeledList>
+      <Box mt="0.5em">
+        <Knob
+          mr="0.5em"
+          animated
+          size={1.25}
+          inline
+          step={5}
+          minValue={0}
+          maxValue={999}
+          value={inputNumber}
+          onDrag={(e, setInput) => act('setInput', { setInput })} />
+        <Button
+          content={'W'}
+          selected={inputMultiplier===1}
+          onClick={() => act('inputW')} />
+        <Button
+          content={'kW'}
+          selected={inputMultiplier===10**3}
+          onClick={() => act('inputkW')} />
+        <Button
+          content={'MW'}
+          selected={inputMultiplier===10**6}
+          onClick={() => act('inputMW')} />
+        <Button
+          content={'GW'}
+          selected={inputMultiplier===10**9}
+          onClick={() => act('inputGW')} />
+        <Button
+          content={'TW'}
+          selected={inputMultiplier===10**12}
+          onClick={() => act('inputTW')} />
+      </Box>
+    </Section>
+  );
+};
+
+const OutputControls = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    isFiring,
+    laserEnabled,
```

Comment:
Mix of variable naming convention here (i.e. one using "is", one not) seems meh. Not requiring you change it, just highlighting code style stuff that we'll never be able to lint so might as well spread the knowledge.

I have no problem with `firing` as a key on `data`.

Reply:
vars have been renamed to `isLaserEnabled` and `isChargingEnabled` to match `isFiring` and `isCharging`